openedx/core/djangoapps/credentials/utils.py

```diff
@@ -64,3 +64,34 @@ def get_user_program_credentials(user):
         programs_credentials_data = get_programs_for_credentials(user, programs_credentials)
 
     return programs_credentials_data
+
+
+def get_programs_credentials(user, category='xseries'):
```

Comment:
The `category` should default to `None` for no filtering. You can provide `xseries` when calling this from the listing view.